internal/orchestrator/sketch_libs.go

```diff
@@ -43,6 +48,16 @@ func AddSketchLibrary(ctx context.Context, app app.ArduinoApp, libRef LibraryRel
 		return nil, err
 	}
 
+	// update the local library_index if it is older than a certain threshold, to ensure the library is found when added by the arduino-cli
+	stream, res := commands.UpdateLibrariesIndexStreamResponseToCallbackFunction(ctx, func(curr *rpc.DownloadProgress) {
+		slog.Debug("downloading library index", "progress", curr.Message)
+	})
+	req := &rpc.UpdateLibrariesIndexRequest{Instance: inst, UpdateIfOlderThanSecs: int64(indexUpdateInterval.Seconds())}
+	if err := srv.UpdateLibrariesIndex(req, stream); err != nil {
+		return []LibraryReleaseID{}, fmt.Errorf("error updating library index: %v", err)
```

Comment:
What happens if the board is disconnected? I think we should allow adding a library also without an internet connection

Reply:
You will need the connection anyway to download the library.